modules/tribes/tests/client/components/TribesPage.component.test.js

```diff
@@ -0,0 +1,177 @@
+/* tests to write
+
+when clicking join and not logged in, redirect to tribe page
+when clicking join and logged in, send a request to join to API
+when clicking join and logged in, update the number of tribe members
+when clicking leave, send a request to leave to API
+when clicking leave and logged in, update the number of tribe members
+
+
+*/
+
+import React from 'react';
+import * as rtl from '@testing-library/react';
+import { render, fireEvent, wait } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import '@/config/client/i18n';
+
+import TribesPage from '@/modules/tribes/client/components/TribesPage.component';
+import * as api from '@/modules/tribes/client/api/tribes.api';
```

Comment:
I was creating mock `api` objects as they are used in the components, e.g.:

```js
import * as tribes from '@/modules/tribes/client/api/tribes.api';
const api = { tribes };
```

Which I liked because:
* maybe some tests need to use two different APIs in some places
* consistent with how it looks in the code where it is used for real
* made it very clear when reading the code later which API I was using

Doesn't seem essential though, kind of stylistic. What think you?

Reply:
It sounds fair to have this consistent. Changed in 48216ee1f3691affb82eca49b6aaca06ed1271be.